### PR TITLE
Exclusively use the describeRoute decorator

### PR DESCRIPTION
**Pull request by brianhelba**
* Update all route handlers to use ``@describeRoute``
* Update docs to only refer to ``@describeRoute`` for applying ``Descriptions``
 * This makes directly setting the ``.description`` property on routes fully deprecated and undocumented (though it will still work).
* Make a test case use the proper type of docstring

### CODE REVIEW

**Comment by zachmullen** [comment]
@brianhelba It seems the `describeRoute(None)` is not having the intended effect. When I restart the server, I'm getting ``WARNING: No description docs present for route GET oauth/:provider/callback``.

File: girder/api/rest.py [hunk 1]
```diff
@@ -578,11 +578,10 @@ def route(self, method, route, handler, nodoc=False, resource=None):
...
 
-        if hasattr(handler, 'description'):
-            if handler.description is not None:
-                docs.addRouteDocs(
-                    resource=resource, route=route, method=method,
-                    info=handler.description.asDict(), handler=handler)
+        if getattr(handler, 'description', None) is not None:
+            docs.addRouteDocs(
+                resource=resource, route=route, method=method,
+                info=handler.description.asDict(), handler=handler)
```
Comment (zachmullen) [inline girder/api/rest.py]: This change is the source of the bug I mentioned.

**Comment by brianhelba** [comment]
@zachmullen Thanks, I'll fix.

Sorry, I intended to mark this as WIP until I got a chance to test it locally. It was was just nagging me the other day, so I made the fixes while waiting for some long computations to finish. I also need to review it for any copy-paste errors.

**Comment by brianhelba** [comment]
@girder/developers PTAL.

**Comment by zachmullen** [comment]
LGTM

**Pull request merged**